api/v1alpha1/httproute_types.go

```diff
@@ -264,6 +264,59 @@ type ForwardToTarget struct {
 	//
 	// +optional
 	TargetPort *TargetPort `json:"targetPort" protobuf:"bytes,2,opt,name=targetPort"`
+
+	// TimeoutPolicy describes a policy for configuring connection timeouts to targetRef.
```

Comment:
I am not sure how much we want to cater to specific implementations but I don't think Envoy supports this, specifically the timeout and retry config is at the route level, so you cannot set it differently for each weighted cluster destination.